src/lib/journey/callbacks/hidden-value/hidden-value.mock.ts

```diff
@@ -1,3 +1,31 @@
+  callbacks: [
+    {
+      type: CallbackType.HiddenValueCallback,
+      output: [
+        {
+          name: 'value',
+          value: 'false',
+        },
+        {
+          name: 'id',
+          value: 'webAuthnOutcome',
```

Comment:
Taking a quick look at the AM codebase, I can see that the following nodes could return a HiddenValueCallback:

- PushWait
- PushRegistration
- WebAuthn
- WebAuthn Registration

... and of course any scripted nodes that could return one.

I chose WebAuthn here, though it will be difficult to test this callback properly and be confident we have implemented everything, until we implement support for these other node callbacks.

Reply:
It's worth noting that the SDK doesn't actually execute the script type from the WebAuthn nodes. [It parses the script to extract the values](https://github.com/ForgeRock/forgerock-javascript-sdk/blob/master/packages/javascript-sdk/src/fr-webauthn/script-parser.ts) and then internally handles it like the JSON version. 

As for the Push nodes, this is for the QR code display. I'd rather "natively" support QR codes, rather than have the Widget actually run the external script. So, we will either parse the script like the WebAuthn or add a switch to send JSON rather than a script from the node.

My long term vision for the Widget is for us to educate the customer on adding native support for the features they want by extending the Widget, rather than writing the scripts in AM and having the Widget execute them. From an architectural standpoint, supporting the features in the Widget is far superior from a quality control and security standpoint. 

I would highly recommend we treat the execution of scripts from AM within client apps to be a legacy method, and provide this superior method and habituate customers to it, discouraging the "legacy" method. With that said, I would actually be okay with this being a "noop" component as we just need a component to exist, so Svelte doesn't get tripped up.

From a WebAuthn callback/feature perspective, the SDK just directly sets the value on the callback internally, and NOT through any hidden DOM input.

Reply:
I think I understand, and aside from the concern in the other comment about partner nodes, I'm happy that we don't need to render anything. 

To clarify, when you say the following:

> With that said, I would actually be okay with this being a "noop" component as we just need a component to exist, so Svelte doesn't get tripped up.

... Are you saying that you think we should render something anyway, or just that we have a Svelte component but don't render anything?

Reply:
I'm actually not certain what the minimum requirement is for a Svelte component. Is a Svelte component valid with no HTML? I honestly don't know. If it needs some kind of HTML, we could just render an empty span tag?

From a type perspective, I would think it at least needs the same required, exported props, as the rest of the callback components to ensure the APIs line up. In other words, these:

1. `callback`
2. `firstInvalidInput`
3. `idx`

Reply:
Hmm good point - as far as I can tell, a Svelte component can be empty and still be valid. My assumption is based on an [article](https://dev.to/jesseskinner/the-simplest-svelte-component-is-an-empty-file-2lhk) and something I saw in the [docs](https://svelte.dev/docs):

> All three sections — script, styles and markup — are optional.

I tested this theory by refactoring the hidden value component to the following, and didn't get yelled at by my IDE, storybook or the test app!

Reply:
Perfect. Let's keep this dead simple, and make decisions about any additional handling when they come up in testing.